test/function_compilation.jl

```diff
@@ -0,0 +1,191 @@
+module TestFunctionCompilation
+
+using Test
+using DataFramesMeta
+
+@testset "function_compilation" begin
+	# Lazy way of making sure all
+	# functions are pre-compiled. @eval
+	# prevents julia from caching the
+	# intermediate anonymous functions.
+	for _ in 1:2
+		@eval begin
+			df = DataFrame(a = [1], b = [2])
+
+			testfun(x, y) = x .* y
+			testdotfun(x, y) = x * y
+			testnt(x) = (c = x,)
+
+			@test @select(df, c = :a + :b) == DataFrame(c = [3])
+
+				fasttime = @timed @select(df, c = :a + :b)
+				slowtime = @timed select(df, [:a, :b] => ((a, b) -> a + b) => :c)
+				@test slowtime[2] > fasttime[2]
```

Comment:
I fixed the indentation and the tabs. 

However looking at `bytes` also errors sometimes in unpredictable ways. 

I wish there were a way to do a `@test` where it logs the result of the test but doesn't complain on Github if it doesn't pass. 

I also had to add an extra `@with` call because somehow it wasn't picking up the compilation. 

So not sure what the solution is.

Reply:
Well you can just do `slowtime[2] > fasttime[2] || @warn(...)`, right? :-)

Reply:
added.